vortex-array/src/array/constant/compute.rs

```diff
@@ -59,11 +70,59 @@ impl SearchSortedFn for ConstantArray {
     }
 }
 
+impl AndFn for ConstantArray {
+    fn and(&self, array: &Array) -> VortexResult<Array> {
+        constant_array_bool_impl(self, array, |(l, r)| l & r, and)
+    }
+}
+
+impl OrFn for ConstantArray {
+    fn or(&self, array: &Array) -> VortexResult<Array> {
+        constant_array_bool_impl(self, array, |(l, r)| l | r, or)
+    }
+}
+
+fn constant_array_bool_impl(
+    constant_array: &ConstantArray,
+    other: &Array,
+    bool_op: impl Fn((bool, bool)) -> bool,
+    fallback_fn: impl Fn(&Array, &Array) -> VortexResult<Array>,
+) -> VortexResult<Array> {
+    if constant_array.len() != other.len() {
+        vortex_bail!("Boolean operations aren't supported on arrays of different lengths")
+    }
+
+    if !constant_array.dtype().is_boolean() || !other.dtype().is_boolean() {
+        vortex_bail!("Boolean operations are only supported on boolean arrays")
+    }
+
+    // If the right side is constant
+    if let Some(true) = other.statistics().compute_is_constant() {
+        let lhs = constant_array.scalar().value().as_bool()?;
+        let rhs = scalar_at(other, 0)?.value().as_bool()?;
+
+        let scalar = match lhs.zip(rhs).map(bool_op) {
+            Some(b) => Scalar::bool(b, Nullability::Nullable),
```

Comment:
This should match the nullability of lhs

Reply:
discussed this point - having consistent nullability makes sense here.